### ISSUE TITLE
Asciidoc's `include` macro is parsed as link

### ISSUE THREAD

**Issue by xtaran** [GitbookIO/gitbook#1015]
I have a book which is spread over many files. I need to be able to build with normal `asciidoc` or `asciidoctor` into a single file, too. I hence use statements like `include::was-ist-debian.adoc[]`.

Gitbook shows this statement as `<p><a href="was-ist-debian.html" class="bare">was-ist-debian.adoc</a></p>` though. I would have expected that it includes the given file at this place instead of placing a link or completely ignored at all (in case it needs to be listed in SUMMARY.adoc separately).

I even tried adding a title parameter (locally) to make it look nicer, but no success: `include::was-ist-debian.adoc[Was ist Debian?]`

(The according code can be found in the gitbook feature branch of dpmb/dpmb at https://github.com/dpmb/dpmb/blob/gitbook/konzepte/linux-dschungel/linux-dschungel.adoc, the book on Gitbook is currently private due to its unfinished Gitbook support at https://dpmb.gitbooks.io/debian-paketmanagement/content/v/gitbook/konzepte/linux-dschungel/linux-dschungel.html)

**Comment by SamyPesse** [comment]
We dont' support `include` in the AsciiDoc syntax. There is a PR for this (https://github.com/GitbookIO/gitbook-asciidoc/pull/6), but it doesn't work (except for tests) because AsciiDoc doesn't have access to the path to the current file.

You can use GitBook conrefs: https://help.gitbook.com/format/conrefs.html

I'll take a look at how to correctly support asciidoc.

**Comment by xtaran** [comment]
Thanks for all these hints. I've subscribed to GitbookIO/gitbook-asciidoc#6 and GitbookIO/gitbook-asciidoc#5.

Unfortunately the content references won't work with plain Asciidoc as far as I can see, so they're currently of no use for me.

I already have a [small Perl script which generates `SUMMARY.adoc` from all these `include::` directives](https://github.com/dpmb/dpmb/blob/gitbook/bin/generate-gitbook-summary.pl). So if I manage to split up all files into either only containing real content _or_ only `include::` statements, it would probably work, too.

Anyway, having `include::` statements supported would still ease a lot, especially if they would be treated similar to the `. link:` macros inside `SUMMARY.adoc`.

**Comment by prudhomm** [comment]
:+1: I submitted a similar issue through zendesk.
that would be definitely an important feature

**Comment by nagim** [comment]
It would be also important to enable partial reuse ([example](http://mrhaki.blogspot.hu/2014/04/awesome-asciidoc-include-partial-parts.html)). I don't think that this kind of reuse is supported in nunjucks, so it would be important to support AsciiDoc's native include method.

**Comment by prudhomm** [comment]
Indeed, lines selection is not very useful for us, tags are much more useful !
gitbook should support this AsciiDoc feature

**Comment by klakegg** [comment]
It's written "because AsciiDoc doesn't have access to the path to the current file". I think path to the current file doesn't matter, all includes should include based on book source folder. Could that make it easier to implement?

Include in adoc is one of the mayor reasons for me to choose Gitbook.

**Comment by lwchkg** [comment]
> It's written "because AsciiDoc doesn't have access to the path to the current file". I think path to the current file doesn't matter, all includes should include based on book source folder. Could that make it easier to implement?

Disagree. Using relative paths is what HTML excels, and we need to add this to GitBook too.
(You can compare with Latex. It produces excellent PDF output, but being unable to use relative paths makes the document structure unmaintainable. Every time I need to move files around I need to edit the files to change the paths!)

BTW, back to solving the issue: there is now a base_dir options in AsciiDoctor.js, see
https://github.com/asciidoctor/asciidoctor.js/issues/197
for how to use it.

And I can write a few pull requests if you want.

**Comment by jaredmorgs** [comment]
This really needs to be addressed. 

If a user evaluating GitBook sees that it has "AsciiDoc" support, then they would instinctively try to use the `include` directive. 

It's one of the power features of AsciiDoc and why folks choose the language over Markdown. Not having `include` support as intended by upstream Asciidoc in GitBook is an Achilles Heel.

**Comment by ZihoRo** [comment]
[gitbook-plugin-asciidoc-include](https://github.com/ZihoRo/gitbook-plugin-asciidoc-include)
